src/discovery/utils.ts

```diff
@@ -0,0 +1,19 @@
+import type { DiscoveryQueueId, DiscoveryQueueIdGenerator } from './types';
+import { IdSortable } from '@matrixai/id';
+import { makeId } from '../GenericIdTypes';
+
+function makeDiscoveryQueueId(arg: any) {
+  return makeId<DiscoveryQueueId>(arg);
+}
+
+function createDiscoveryQueueIdGenerator(
+  lastId?: DiscoveryQueueId,
+): DiscoveryQueueIdGenerator {
+  const idSortableGenerator = new IdSortable({
```

Comment:
Is there a reason why the `DiscoveryQueueId` is an `IdSortable`?

Reply:
It's a time-ordered ID stored in the DB. That's the primary reason. It's also monotonic, and you can also extract the timestamp out of it.

Reply:
Yes, that's correct. The usage is exactly the same as in `NotificationsManager` with the `NotificationId` there.

Reply:
Great - resolving.